drivers/nrf_802154/driver/src/mac_features/nrf_802154_ie_writer.c

```diff
@@ -69,36 +76,87 @@ static uint16_t  m_csl_period;          ///< CSL period value that will be injec
 static uint64_t  m_csl_anchor_time;     ///< The anchor time based on which CSL window times are calculated
 static bool      m_csl_anchor_time_set; ///< Information if CSL anchor time was set by the higher layer
 
-static bool csl_time_to_nearest_window_midpoint_get(uint32_t * p_time_to_midpoint)
+#if defined(CONFIG_SOC_SERIES_BSIM_NRFXX)
+static uint32_t m_csl_time_to_radio_address_us;
+
+#endif
+
+/** @brief Calulate CSL phase.
+ *
+ * @param[out]  p_csl_phase   Calculated CSL phase in units of 160us.
+ *
+ * @retval  true   The calculation was successful and @p p_csl_phase contains a valid CSL phase.
+ * @retval  false  The calculation failed and the value pointed to by @p p_csl_phase is undefined.
+ */
+static bool csl_phase_calc(uint32_t * p_csl_phase)
 {
-    bool result = false;
+    bool     result = false;
+    uint32_t us;
 
     if (m_csl_anchor_time_set)
     {
         result = (m_csl_period != 0);
 
         if (result)
         {
-            uint64_t now           = nrf_802154_sl_timer_current_time_get();
-            uint32_t csl_period_us = m_csl_period * IE_CSL_SYMBOLS_PER_UNIT * PHY_US_PER_SYMBOL;
+            /*
+             * This function is executed in the handler of RADIO.ADDRESS event. According to the IPS,
+             * in 802.15.4 transmit sequence RADIO.FRAMESTART event is triggered after the SHR is
+             * transmitted (nRF52840 PS v1.7 -- 6.20.12.6 Transmit sequence). However, RADIO.ADDRESS
+             * event is also triggered in 802.15.4 transmit sequence with a constant 32us offset.
+             * This handler is therefore expected to execute 32us before the SHR transmission ends.
+             *
+             * The IEEE 802.15.4-2015 specification is unclear about which point in time the CSL Phase
+             * should refer to. Following the latest developments in the Thread specification, here
```

Comment:
Hi @fgrandel, the clarification about CSL Phase has been performed in the context of the SPEC-1154 Thread Group Jira issue, affecting the Thread Specification section 3.2.6.3.4. Only Thread Group members have access to this information (yet).

Reply:
@edmont Ok, makes sense. That's why I suggested to document in the [public API](https://github.com/zephyrproject-rtos/hal_nordic/pull/136#discussion_r1244253561) how channel samples (RX windows) are derived from anchor points. It might not be so obvious for most users that anchor points are offset with respect to channel sample midpoints. (But then how many users care anyway?)